src/cryptonote_core/blockchain.cpp

```diff
@@ -1219,44 +1265,47 @@ difficulty_type Blockchain::get_difficulty_for_alternative_chain(const std::list
 bool Blockchain::prevalidate_miner_transaction(const block& b, uint64_t height, uint8_t hf_version)
 {
   LOG_PRINT_L3("Blockchain::" << __func__);
-  CHECK_AND_ASSERT_MES(b.miner_tx.vin.size() == 1, false, "coinbase transaction in the block has no inputs");
-  CHECK_AND_ASSERT_MES(std::holds_alternative<txin_gen>(b.miner_tx.vin[0]), false, "coinbase transaction in the block has the wrong type");
-  if (var::get<txin_gen>(b.miner_tx.vin[0]).height != height)
+  if (b.miner_tx.vout.size() > 0)
```

Comment:
micro optimization for the future: `not vec.empty()` is often slightly faster than `vec.size() > 0` because cache. it is also much nicer to read.